torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -616,6 +626,72 @@ std::vector<at::Tensor> flatten_for_scatter_gather(
   return flattened;
 }
 
+void checkSplitSizes(
```

Comment:
This is a duplication of the `checkSplitSizes` in `ProcessGroupMPI.cpp`? If so, let's move it to some common place, e.g., a protected method in `ProcessGroup`, to avoid duplication..